### ISSUE TITLE
BUILD: the future of build/deploy for Xemu aka. Travis problems

### ISSUE THREAD

**Issue by lgblgblgb** [lgblgblgb/xemu#211]
Xemu uses the Travis service to build and deploy official builds of Xemu, quite effortlessly. This is a very important point, as it would be hard to maintain that manually, to build for all the supported platforms (especially for Mac, since I don't have any Mac, or access to any Mac).

Travis has announced, travis.org will be shut down (near the end of year 2020?). They asks users to migrate their works over travis.com. Though I am not sure, why they can't do this automatically, anyway I should do it, to provide uninterrupted flow of official builds for the Xemu project.

**Comment by lgblgblgb** [comment]
Interesting, now travis says: "_Please be aware travis-ci.org will be shutting down in several weeks, with all accounts migrating to travis-ci.com. Please stay tuned here for more information."_ So then, will it be an automatic process and I don't need to do the migration??

**Comment by lgblgblgb** [comment]
Travis seems to be confused a lot, once they mention "all accounts migrating" (ie, they do it), now they states: the deadline is moved further since so many people hasn't moved yet ... Now what? They move the users, or users needs to move at their own?!

**Comment by lgblgblgb** [comment]
Ok, `travis-ci.org` has been closed without moving the project as they **promised** earlier. Really fantastic ... Anyway, trying `travis-ci.com` now with manual migration ... If everything works out OK, I can close this one.

**Comment by lgblgblgb** [comment]
It seems `travis-ci.com` has strange policy. You must write mail to "ask" OSS credits. And MacOS counts more "bad" than Linux in terms of credit usage. "payed" credits are given for the free plan as well (10'000) however it seems a single offer without extra payment. Should be considered to use `github actions` or at least check which suits better.

**Comment by lgblgblgb** [comment]
Some annoying random info:
https://news.ycombinator.com/item?id=25338983
https://www.theregister.com/2020/11/02/travis_ci_pricng/

**Comment by lgblgblgb** [comment]
Current situation, I had to answer questions and "prove" things, I got 25K credits, but **not** OSS credits, regular ones. Strange (since the OSS credits can be spent only on OSS things ... why would they give me "normal" ones?). Also it seems I have to renew, if running out (and surely then can say "no" on re-review). Sounds a bit scary for me.

Let's stick with Travis for now, till I have some major issue with this. Meanwhile, it would be great, to abstract official building with intermediate scripts, thus it can be adopted to call from different CI provider, or anything, without the current situation where many travis specific things are "wired in".

**Comment by lgblgblgb** [comment]
**NOTE TO MYSELF**: I have to refer for ticket "Travis CI: [29741] - Requesting OSS credits" later if I want to "re-new" credits, and not needed to have the review again, if I do this way. It seems.

**Comment by lgblgblgb** [comment]
Re-opened for an on-going issue, involving Travis again but with different problem:

So far I've used Ubuntu 20.04 ("focal") on Travis to compile for Linux and Windows (using mingw cross compiler) as well. However the included gcc and mingw-gcc is aging on that image, so I've tried to use a more up-to-date Ubuntu 22.04 ("jammy"). The build itself went well, however some Travis related ruby problem happened.

* My report on travis: https://travis-ci.community/t/ubuntu-jammy-installing-deploy-dependencies-failed-to-deploy/13947
* Similar report from others: https://travis-ci.community/t/github-release-fail-after-ruby-version-update/13214
